src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -635,7 +631,8 @@ void Assembler::emit_operand_helper(int reg_enc, int base_enc, int index_enc,
                        scale, index_enc, base_enc);
         emit_data(disp, rspec, disp32_operand);
       }
-    } else if (base_enc == rsp->encoding() LP64_ONLY(|| base_enc == r12->encoding())) {
+    } else if (base_enc == rsp->encoding() LP64_ONLY(|| base_enc == r12->encoding()
+              || base_enc == 20 || base_enc == 28)) {
```

Comment:
This could be simplified to:
if ((base_enc & 0x7) == 4) {

Reply:
Done. Thanks.